src/domains/agents/communication/proactive_agent.py

```
@@ -273,10 +333,58 @@ async def _generate_proactive_message(self, message_type: str, belief_content: D
-            user_name = user_context.get("name", "there")
+            user_name = context.get("user_name", user_context.get("name", "there"))
+            user_message = context.get("user_message", "")
+            
+            logger.info(f"Generating message for type: {message_type}, user: {user_name}, message: {user_message}")
             
-            # Build context-specific prompt
-            if message_type == "send_meeting_confirmation":
+            # Handle conversational chat with rich context
+            if message_type == "conversational_chat" or context.get("conversation_type") == "natural_conversation":
+                conversation_history = context.get("conversation_history", [])
+                user_profile = context.get("user_profile", {})
```

Ruff (0.12.2)

Local variable `user_profile` is assigned to but never used

Remove assignment to unused variable `user_profile`

(F841)